### PR TITLE
feat: jenkins Plugin jenkins_url Improvement

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [ ] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description

jenkins plugin jenkins_url improvement

## Related Issues

https://github.com/devstream-io/devstream/pull/1025#discussion_r952277174

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

File: internal/pkg/plugin/jenkins/jenkins.go [hunk 1]
```diff
@@ -29,20 +32,46 @@ func genJenkinsState(options plugininstaller.RawOptions) (statemanager.ResourceS
...
+	var retStr string
+	if strings.Contains(opts.Chart.ChartName, opts.Chart.ReleaseName) {
+		retStr = strings.TrimSuffix(opts.Chart.ReleaseName, "-")
+		if len(retStr) > 63 {
+			retStr = retStr[:64]
+		}
+		return retStr, nil
+	}
+	retStr = fmt.Sprintf("%s-%s", opts.Chart.ReleaseName, opts.Chart.ChartName)
+
```
Comment (iyear) [inline internal/pkg/plugin/jenkins/jenkins.go:75]: image

From the pic, the `if` and `else` branches all require `trunc 63` and `trimSuffix`.

Comment (iyear) [reply internal/pkg/plugin/jenkins/jenkins.go:75]: or may be written like this:
```go
pipe := func(s string) string {
	if len(s) > 63 {
		s = s[:64]
	}
	return strings.TrimSuffix(s, "-")
}

if strings.Contains(opts.Chart.ChartName, opts.Chart.ReleaseName) {
	return pipe(opts.Chart.ReleaseName), nil
}
return pipe(fmt.Sprintf("%s-%s", opts.Chart.ReleaseName, opts.Chart.ChartName)), nil
```

Comment (daniel-hutao) [reply internal/pkg/plugin/jenkins/jenkins.go:75]: Good tips. But I think the original requirements are:

1. length <=63
2. suffix != "-"

So trim suffix first is better.

Comment (iyear) [reply internal/pkg/plugin/jenkins/jenkins.go:75]: But if the last character after you trim is `-`

Comment (iyear) [reply internal/pkg/plugin/jenkins/jenkins.go:75]: `hell- world`
1. trunc 5 | trim => `hell`
2. trim | trunc 5 => `hell-`

Comment (daniel-hutao) [reply internal/pkg/plugin/jenkins/jenkins.go:75]: Yeah, I agree with you. I'll fix it asap.

**Review by iyear — APPROVED** [review]
LGTM

**Review by aFlyBird0 — APPROVED** [review]
LGTM 🚀

**Pull request merged**